src/main/resources/static/account.html

```diff
@@ -49,19 +50,26 @@ <h2>Select your avatar</h2>
 
     document.getElementById("confirmChanges").onclick = () => {
 
-        fetch("/api/auth/account",
-            {
-                method: "POST",
-                headers: {
-                    "Content-Type": "application/json",
-                    'Authorization': 'bearer ' + token,
-                },
-                body: JSON.stringify({
-                    email: Auth.getEmail(),
-                    accountName: document.getElementById("username").value,
-                    avatar: sessionStorage.getItem('byte64Image')
-                })
-            })
+        if (document.getElementById("username").value === "") {
+                document.getElementById("usernameEmpty").innerText = "username cannot be empty"
+                } else {
+
+                        document.getElementById("usernameEmpty").innerText = ""
+                    fetch("/api/auth/account",
+                        {
+                            method: "POST",
+                            headers: {
+                                "Content-Type": "application/json",
+                                'Authorization': 'bearer ' + token,
+                            },
+                            body: JSON.stringify({
+                                email: Auth.getEmail(),
+                                accountName: document.getElementById("username").value,
+                                avatar: sessionStorage.getItem('byte64Image')
+                            })
+                        })
+                        location.reload();
```

Comment:
I didn't have luck with it, but I imagine this line might be executed before fetch is sent, which could stop request from being sent. 
fetch doesn't stop program and doesn't wait for completion of request. I'd suggest reloading in `then()`